src/vec.rs

```diff
@@ -1140,27 +1140,29 @@ impl<T: Clone> ExactSizeIterator for IntoIter<T> {}
 
 impl<T> Drop for IntoIter<T> {
     fn drop(&mut self) {
-        if self.unique && self.vec.is_allocated() {
+        if !(self.unique && self.vec.is_allocated()) {
```

Comment:
Here I agree that the early return is better. But could you De-Morgan the condition to `!self.unique || !self.vec.is_allocated()`?

Reply:
Resolved in latest commit